### PR TITLE
Add tests for each cert type

### PR DESCRIPTION
**Pull request by rain-on**
This required updates to the DSL to allow for conversion of
certificate types.

### CODE REVIEW

File: acceptance-tests/src/test/java/tech/pegasys/signers/keystorage/hashicorp/tests/HashicorpVaultAccessAcceptanceTest.java [hunk 1]
```diff
@@ -48,11 +52,16 @@ void cleanup() {
       hashicorpNode.shutdown();
       hashicorpNode = null;
     }
+
+    try {
+      docker.close();
+    } catch (final Exception ignored) {
```
Comment (usmansaleem) [inline acceptance-tests/src/test/java/tech/pegasys/signers/keystorage/hashicorp/tests/HashicorpVaultAccessAcceptanceTest.java]: nit: may be put a system.err or debug log that docker close failed. Empty blocks are frowned upon by some devs :)

File: keystorage/build.gradle [hunk 2]
```diff
@@ -0,0 +1 @@
+jar { enabled = false }
```
Comment (usmansaleem) [inline keystorage/build.gradle]: whats this about?

Comment (rain-on) [reply keystorage/build.gradle]: Its something we should have always had - i.e. we should not create a jar file at this level of the project (no, it shouldn't be in THIS pr) - but it does need to go in sometime.

File: acceptance-tests/src/test/java/tech/pegasys/signers/keystorage/hashicorp/tests/HashicorpVaultAccessAcceptanceTest.java (hunk 1 above)
Comment (rain-on) [reply acceptance-tests/src/test/java/tech/pegasys/signers/keystorage/hashicorp/tests/HashicorpVaultAccessAcceptanceTest.java]: yeah -fixed.

File: acceptance-tests/src/test/java/tech/pegasys/signers/dsl/hashicorp/HashicorpNode.java [hunk 3]
```diff
@@ -94,22 +81,10 @@ public int getPort() {
   }
 
   public boolean isTlsEnabled() {
-    return tlsServerCertificate != null;
-  }
-
-  public Optional<Path> getKnownServerFilePath() {
-    return knownServerFile;
+    return serverTlsCertificate != null;
```
Comment (jframe) [inline acceptance-tests/src/test/java/tech/pegasys/signers/dsl/hashicorp/HashicorpNode.java]: Shouldn't this just be serverTlsCertifcate.isPresent considering it's an optional

File: acceptance-tests/src/test/java/tech/pegasys/signers/dsl/hashicorp/HashicorpNode.java [hunk 4]
```diff
@@ -94,22 +81,10 @@ public int getPort() {
   }
 
   public boolean isTlsEnabled() {
```
Comment (jframe) [inline acceptance-tests/src/test/java/tech/pegasys/signers/dsl/hashicorp/HashicorpNode.java]: Looks like this method isn't used

File: acceptance-tests/src/test/java/tech/pegasys/signers/dsl/certificates/CertificateHelpers.java [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+import java.security.cert.Certificate;
+import java.security.cert.CertificateEncodingException;
+import java.security.cert.CertificateException;
+import java.util.Optional;
+
+import org.apache.tuweni.net.tls.TLS;
+
+public class CertificateHelpers {
+
+  public static void populateFingerprintFile(
```
Comment (jframe) [inline acceptance-tests/src/test/java/tech/pegasys/signers/dsl/certificates/CertificateHelpers.java]: Why does write to a supplied path whereas the other functions return a new file path

File: acceptance-tests/src/test/java/tech/pegasys/signers/keystorage/hashicorp/tests/HashicorpVaultAccessAcceptanceTest.java [hunk 6]
```diff
@@ -89,13 +114,97 @@ void keyCanBeExtractedFromVaultOverTlsUsingWhitelist() throws IOException {
...
+  void canConnectToHashicorpVaultUsingPkcs12Certificate(@TempDir final Path testDir)
+      throws IOException {
+    final String TRUST_STORE_PASSWORD = "password";
+    hashicorpNode = HashicorpNode.createAndStartHashicorp(docker, true);
+
+    final Path trustStorePath =
+        CertificateHelpers.createPkcs12TrustStore(
+            testDir, hashicorpNode.getServerCertificate().get(), TRUST_STORE_PASSWORD);
+
+    // create tomlfile
```
Comment (jframe) [inline acceptance-tests/src/test/java/tech/pegasys/signers/keystorage/hashicorp/tests/HashicorpVaultAccessAcceptanceTest.java]: nit: perhaps remove the comment about this being a toml file. I don't think we care it's a config file

File: acceptance-tests/src/test/java/tech/pegasys/signers/dsl/hashicorp/HashicorpNode.java (hunk 4 above)
Comment (rain-on) [reply acceptance-tests/src/test/java/tech/pegasys/signers/dsl/hashicorp/HashicorpNode.java]: done

File: acceptance-tests/src/test/java/tech/pegasys/signers/keystorage/hashicorp/tests/HashicorpVaultAccessAcceptanceTest.java (hunk 6 above)
Comment (rain-on) [reply acceptance-tests/src/test/java/tech/pegasys/signers/keystorage/hashicorp/tests/HashicorpVaultAccessAcceptanceTest.java]: done

**Pull request merged**